rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -2942,6 +2942,21 @@ public String loginOrRenewSession(@HeaderParam("sessionid") String sessionId,
     }
 
     /**
+     * {@inheritDoc}
+     */
+    @Override
+    @GET
+    @Path("logins/sessionid/{sessionId}")
+    @Produces("application/json")
+    public String getLoginFromSessionId(@PathParam("sessionId") String sessionId)
+            throws SchedulerRestException, LoginException, NotConnectedRestException {
+        if (sessionId != null && sessionStore.exists(sessionId)) {
+            return sessionStore.get(sessionId).getUserName();
+        }
+        return "";
```

Comment:
i dont think so, if there is not a login associated with the given session id, simply return empty, meaning that the login resource doesn't exist, i dont think it is good to throw an exception if so.